tests/unit/ProcessMaker/InboxRules/MatchingTasksTest.php

```diff
@@ -75,24 +76,38 @@ public function testInboxRuleEndDate()
         // not be returned
 
         $user = User::factory()->create();
-        $completedTask = ProcessRequestToken::factory()->create([
+    
+        $task = ProcessRequestToken::factory()->create([
             'user_id' => $user->id,
-            'status' => 'COMPLETED',
+            'status' => 'ACTIVE',
         ]);
-        $activeTask = ProcessRequestToken::factory()->create([
+
+        $futureEndDate = now()->addDays(1);
+        
+        $pastEndDate = now()->subDays(1);
+        InboxRule::factory()->create([
+            'process_request_token_id' => $task->id,
+            'end_date' => $pastEndDate,
             'user_id' => $user->id,
-            'status' => 'ACTIVE',
-            'element_id' => $completedTask->element_id,
-            'process_id' => $completedTask->process_id,
         ]);
-        $inboxRule = InboxRule::factory()->create([
-            'process_request_token_id' => $completedTask->id,
+
+        $matchingRules = MatchingTasks::matchingInboxRules($task);
+
+        $expectinginboxRule = InboxRule::factory()->create([
+            'process_request_token_id' => $task->id,
+            'end_date' => $futureEndDate,
             'user_id' => $user->id,
         ]);
 
-        $matchingTasks = MatchingTasks::matchingInboxRules($activeTask);
+        $notMatchingRules = MatchingTasks::matchingInboxRules($task);
 
-        $this->assertCount(1, $matchingTasks);
-        $this->assertEquals($inboxRule->id, $matchingTasks[0]->id);
+        $this->assertEmpty($matchingRules);
+        $this->assertNotEmpty($notMatchingRules);
+        //when end date is not past and we have a task match
+        $this->assertEquals(
+            $expectinginboxRule->process_request_token_id,
+            $notMatchingRules[0]->process_request_token_id
+        );
+        
```

Comment:
Try this instead. Only create 1 inbox rule for the test. First, set it to the pastEndDate and assert there are no matches. Then, $inboxRule->update to set the future end date. Then run MatchingTasks::matchingInboxRules again and assert it has 1 matching inbox rule

Reply:
Solved